katdal/visdatav4.py

```diff
@@ -564,14 +564,19 @@ def _set_keep(self, time_keep=None, freq_keep=None, corrprod_keep=None,
                 self._excision = None
             else:
                 # The maximum / expected number of CBF dumps per SDP dump
-                cbf_dumps_per_sdp_dump = int(round(self.dump_period / self.cbf_dump_period))
+                cbf_dumps_per_sdp_dump = round(self.dump_period / self.cbf_dump_period)
                 accs_per_sdp_dump = np.float32(self.accumulations_per_dump)
                 accs_per_cbf_dump = accs_per_sdp_dump / np.float32(cbf_dumps_per_sdp_dump)
                 # Each unscaled weight represents the actual number of accumulations per SDP dump.
                 # Correct most of the weight compression artefacts by forcing each weight to be
                 # an integer multiple of CBF n_accs, and then convert it to an excision fraction.
-                excision_transforms = [lambda w: da.round(w / accs_per_cbf_dump) * accs_per_cbf_dump,
-                                       lambda w: (accs_per_sdp_dump - w) / accs_per_sdp_dump]
+                def integer_cbf_dumps(w):
+                    return da.round(w / accs_per_cbf_dump) * accs_per_cbf_dump
+
+                def excision_fraction(w):
+                    return (accs_per_sdp_dump - w) / accs_per_sdp_dump
+
+                excision_transforms = [integer_cbf_dumps, excision_fraction]
```

Comment:
I'd been thinking this would be a single transformation function. What are the performance implications of doing the transformation in two steps?

Reply:
I'll have to benchmark it... I was careful to keep everything float32 as the first priority. If I look at the dask graph I see 5 operations (truediv, round, mul, subtract, truediv) so the functions are unpacked anyway. Combining the functions might get rid of one mul operation, but maybe not more.

Reply:
I guess it won't really affect the dask graph whether it is one function or two.